minio/api.py

```diff
@@ -698,6 +698,8 @@ def fget_object(self, bucket_name, object_name, file_path):
         :param bucket_name: Bucket to read object from.
         :param object_name: Name of the object to read.
         :param file_path: Local file path to save the object.
+        :param metadata: Any additional metadata to be uploaded along
```

Comment:
As example I can provide a case, when you using SSE-C and want to provide custom headers to get decrypted file from S3:

```python
    key = b'32byteslongsecretkeymustprovided'
    encryption_key = base64.b64encode(key).decode()
    encryption_key_md5 = base64.b64encode(hashlib.md5(key).digest()).decode()

    minio = Minio('s3.amazonaws.com', access_key=AWSAccessKeyId, secret_key=AWSSecretKey)

    obj = minio.get_object('minio-vfs-test', 'test_crypt.txt', metadata={
        'x-amz-server-side-encryption-customer-algorithm': 'AES256',
        'x-amz-server-side-encryption-customer-key': encryption_key,
        'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
    })

    print(obj.read())

    minio.put_object('minio-vfs-test', 'test_crypt1.txt', content, content.getbuffer().nbytes,
                     metadata={
                         'x-amz-server-side-encryption-customer-algorithm': 'AES256',
                         'x-amz-server-side-encryption-customer-key': encryption_key,
                         'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
                     })

    obj = minio.get_object('minio-vfs-test', 'test_crypt1.txt', metadata={
        'x-amz-server-side-encryption-customer-algorithm': 'AES256',
        'x-amz-server-side-encryption-customer-key': encryption_key,
        'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
    })

    print(obj.read())
```